src/app/app.component.ts

```diff
@@ -9,5 +11,17 @@ import { MatDialog } from '@angular/material/dialog';
 export class AppComponent implements OnInit {
   constructor(public form: MatDialog) {}
 
-  ngOnInit() {}
+  ngOnInit() {
+    if (localStorage.getItem('token') === null) {
+      this.openAuthForm();
+    }
+  }
+
+  openAuthForm() {
+    let formInfo = this.form.open(AuthorizationComponent);
```

Comment:
do you need variable?

Reply:
unusual solution to have auth within modal window, usually we have separate page for it, so let's have auth page instead of window